contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -135,7 +135,10 @@ def _transmit_without_retry(self, envelopes):
             ['dc.services.visualstudio.com'],
         )
         try:
-            response = requests.post(
+            session = requests.Session()
+            # Ensure this request is not captured by standard metrics
+            session.disable_collection = True
```

Comment:
This would only eliminate the requests from azure exporter, and applies to metrics (but not traces/logs) only?

Reply:
This is an interesting topic. We need to define exactly which requests from what sources will be collected. The sources are, the user's application, only the metrics exporter in azure, everything in azure and everything in OpenCensus. With this we can ignore collections in certain ways:

1. Ignore only requests in metrics exporter
2. Ignore requests in Azure Exporter
3. Ignore all requests from OpenCensus (including other ext)

IMO as a user, when I instrument with OC, I would want none of the requests from OC to be tracked. So if I were to instrument with multiple exporters (stackdriver, azure, etc.) I wouldn't want those requests (if any) to be tracked in my metrics. With that being said, opencensus-ext-azure is the only extension that utilizes "requests", so as you mentioned, adding logic for disabling checks for everything in that extension would be needed. 

In the future, if other http libraries need to be supported, the same logic should apply to those.

Reply:
What are your thoughts on this having a more descriptive name with less chance for conflicts? e.g. prefixing with `oc_`?

Reply:
Possible. But I think we are going with the solution of using the RuntimeContext to track whether the request is coming from an exporter thread or not. The variable name will be changed appropriately.